addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/handler/NeeoRoomHandler.java

```diff
@@ -0,0 +1,292 @@
+import java.util.Objects;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.neeo.NeeoBindingConstants;
+import org.openhab.binding.neeo.NeeoUtil;
+import org.openhab.binding.neeo.internal.NeeoBrainApi;
+import org.openhab.binding.neeo.internal.NeeoHandlerCallback;
+import org.openhab.binding.neeo.internal.NeeoRoomProtocol;
+import org.openhab.binding.neeo.internal.StatefulHandlerCallback;
+import org.openhab.binding.neeo.internal.type.UidUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An extension of {@link AbstractThingHandler} that is responsible for handling commands for a room
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+public class NeeoRoomHandler extends AbstractThingHandler {
+
+    /** The logger */
+    private final Logger logger = LoggerFactory.getLogger(NeeoRoomHandler.class);
+
+    /**
+     * The initialization task (null until set by {@link #initializeTask()} and set back to null in {@link #dispose()}
+     */
+    private final AtomicReference<ScheduledFuture<?>> initializationTask = new AtomicReference<ScheduledFuture<?>>(
+            null);
+
+    /**
+     * The refresh task (null until set by {@link #initializeTask()} and set back to null in {@link #dispose()}
+     */
+    private final AtomicReference<ScheduledFuture<?>> refreshTask = new AtomicReference<ScheduledFuture<?>>(null);
+
+    /** The {@link NeeoRoomProtocol} (null until set by {@link #initializationTask}) */
+    private final AtomicReference<NeeoRoomProtocol> roomProtocol = new AtomicReference<NeeoRoomProtocol>();
+
+    /**
+     * Instantiates a new neeo room handler.
+     *
+     * @param thing the non-null thing
+     */
+    public NeeoRoomHandler(@NonNull Thing thing) {
+        super(thing);
+        Objects.requireNonNull(thing, "thing cannot be null");
+    }
+
+    /**
+     * Handles commands sent to the room
+     *
+     * @see
+     *      org.eclipse.smarthome.core.thing.binding.ThingHandler#handleCommand(org.eclipse.smarthome.core.thing.ChannelUID,
+     *      org.eclipse.smarthome.core.types.Command)
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        Objects.requireNonNull(channelUID, "channelUID cannot be null");
+        Objects.requireNonNull(channelUID, "command cannot be null");
+
+        final String[] groupIds = UidUtils.parseGroupId(channelUID);
```

Comment:
Because this isn't the group id you are thinking of.  Typically it's "groupId#id" but mine is "groupid-subgroup-etc#id".  So the openHAB groupid is "groupid-subgroup-etc" but what I'm looking for is the array ["groupid","subgroup","etc"]